packages/ui/src/components/va-switch/VaSwitch.vue

```diff
@@ -251,6 +252,15 @@ export default defineComponent({
     @include va-disabled;
   }
 
+  &--readonly {
+    @include va-readonly;
+
+    .va-switch__label {
+      cursor: initial;
+      pointer-events: auto;
+    }
```

Comment:
Doesn't it inherits from parent `--readonly` class?

Reply:
No. Everything was tested and works as Vitaly suggested - "arrow" cursor for interactive area and "text" cursor for label.